### PR TITLE
Improvement on team registration

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
An improvement was made on the team registration page 
1. Display a success message and redirect the user to the login page if the registration is successful
2.  Display error message if the email has already been registered 
3. check password  === confirmed password 

# Type of change 
- [ ] Bug fix
- [ ] New feature
- [x] Refactoring
- [ ] Documentation 
- [x] Improvement 

# Problem
Briefly describe the root cause and analysis of the problem 

# Solution description
See description 

# Tests 
All unit tests pass? N/A

Builds successfully? Yes

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
image
image

Provide steps in detail for reviewers to reproduce the expected result of the code changes.
Try to register a team and test the feature in description section 

# Relevant document/ link (if any) 
Any related documents could help reviewers to review this PR 

# Risk
Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.
Low

### CODE REVIEW

File: client/src/actions/types.js [hunk 1]
```diff
@@ -1,16 +1,15 @@
 /**
  * This file houses our Action Creator type constants.
  */
-export const AUTH_SIGN_IN_REQUEST = 'AUTH_SIGN_IN_REQUEST';
-export const AUTH_SIGN_IN_SUCCESS = 'AUTH_SIGN_IN_SUCCESS';
-export const AUTH_SIGN_IN_FAIL = 'AUTH_SIGN_IN_FAIL';
-export const AUTH_SIGN_OUT = 'AUTH_SIGN_OUT';
+export const SIGN_IN_REQUEST = 'IGN_IN_REQUEST';
```
Comment (aksh0001) [inline client/src/actions/types.js:4]: Typo on the value of this variable

File: client/src/actions/types.js [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
+export const SIGN_IN_SUCCESS = 'SIGN_IN_SUCCESS';
+export const SIGN_IN_FAIL = 'SIGN_IN_FAIL';
+export const SIGN_OUT = 'SIGN_OUT';
+export const REGISTRATE_SUCCESS = 'REGISTRATE_SUCCESS';
```
Comment (aksh0001) [inline client/src/actions/types.js:8]: Can  replace this to `REGISTER_SUCCESS`/`REGISTRATION_SUCCESS`/similar? Ditto naming convention in other files.

File: client/src/reducers/authReducer.js [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
  */
-import { AUTH_SIGN_IN_REQUEST, AUTH_SIGN_IN_SUCCESS, AUTH_SIGN_IN_FAIL, AUTH_SIGN_OUT } from '../actions/types';
+import { SIGN_IN_REQUEST, SIGN_IN_SUCCESS, SIGN_IN_FAIL, SIGN_OUT, REGISTRATE_SUCCESS } from '../actions/types';
 import Cookies from 'js-cookie';
 
 const INITIAL_AUTH_STATE = { 
   signIn: Cookies.get('isLogin') ? true : false,
-  loading: false
+  loading: false,
+  isRegistrated: null,
```
Comment (aksh0001) [inline client/src/reducers/authReducer.js:10]: `isRegistered`?

File: client/src/components/auth/RegistrationForm.js [hunk 4]
```diff
@@ -2,21 +2,29 @@
...
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { useHistory } from 'react-router-dom';
 import toast from 'react-hot-toast';
 
 export default function RegistrationForm() {
   const dispatch = useDispatch();
   const history = useHistory();
   const [validated, setValidated] = useState(false);
+  const isRegistrated = useSelector(state => state.auth.isRegistrated)
```
Comment (aksh0001) [inline client/src/components/auth/RegistrationForm.js:20]: `isRegistered`?

File: client/src/components/auth/RegistrationForm.js [hunk 5]
```diff
@@ -35,7 +43,10 @@ export default function RegistrationForm() {
     event.preventDefault();
     if (form.checkValidity() === false) {
       event.stopPropagation();
-    } else {
+    } else{
+      if (inputs.password !== inputs.confirmPassword){
+        return toast.error('Passwords does not match');
```
Comment (aksh0001) [inline client/src/components/auth/RegistrationForm.js:48]: nit: `Passwords do not match`

File: client/src/actions/team.js [hunk 6]
```diff
@@ -26,25 +26,9 @@ export const createTeam = (teamInfo) => async (dispatch) => {
   try{
     await api.createTeam(teamInfo)
     dispatch(successMessageCreator("Team has been created"));
```
Comment (aksh0001) [inline client/src/actions/team.js:28]: It isn't too clear why there are two calls made to `dispatch`. Add a comment above each to explain what they're for. E.g. the first dispatch is used to signal that a notification should be shown and the second allows us to know when user has been registered successfully to allow us to go back to the login page.

**Review by aksh0001 — APPROVED** [review]
LGTM, only minor comments to be addressed.

File: client/src/actions/team.js [hunk 7]
```diff
@@ -10,11 +10,11 @@ import {
   CREATE_TEAM_MEMBER,
   UPDATE_TEAM_MEMBER,
   DELETE_TEAM_MEMBER,
-  ADD_TEAM,
   GET_GH_ACCESS_TOKEN,
   DEPLOY_SUCCESS,
   DEPLOY_FAIL,
   UPDATE_TEAM,
+  REGISTRATE_SUCCESS,
```
Comment (rpra0006) [inline client/src/actions/team.js:17]: Might want to change REGISTRATE_SUCCESS as "register" naming convention is used in other files?

File: client/src/actions/team.js [hunk 8]
```diff
@@ -26,25 +26,9 @@ export const createTeam = (teamInfo) => async (dispatch) => {
...
-export const getTeam = (teamCredentials) => async (dispatch) => {
-  try {
-    const data = await api.loginTeam(teamCredentials);
-    const teamData = data.data.team;
-    const team = teamDataAllocator(teamData);
     dispatch({
-      type: ADD_TEAM,
-      payload: team,
-    });
+      type: REGISTRATE_SUCCESS
```
Comment (rpra0006) [inline client/src/actions/team.js:30]: REGISTRATE is used here again

File: client/src/components/auth/RegistrationForm.js [hunk 9]
```diff
@@ -2,21 +2,29 @@
...
  */
 
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import Jumbotron from 'react-bootstrap/Jumbotron';
 import Button from 'react-bootstrap/Button';
 import Form from 'react-bootstrap/Form';
 import { Col } from 'react-bootstrap';
 import { createTeam } from '../../actions/team';
 import './Register.css';
```
Comment (rpra0006) [inline client/src/components/auth/RegistrationForm.js:11]: Example that Register is used as naming convention, file name is also RegistrationForm.

**Review by rpra0006 — APPROVED** [review]
Looks good, just need to be consistent with naming conventions for 'register'.

**Review by vgoh1 — APPROVED** [review]

**Pull request merged**